boards/arm/bt610/bt610.dts

```diff
@@ -60,128 +60,32 @@
-			gpios = <&gpio0 2 GPIO_PULL_UP>;
-		};
-
-		ain2: ain_2 {
-			label = "Analog Input 2";
-			gpios = <&gpio0 3 GPIO_PULL_UP>;
-		};
-
-		vref: v_ref {
-			label = "Voltage Reference";
-			gpios = <&gpio0 29 GPIO_PULL_UP>;
-		};
-
-		expanderreset: expander_reset {
-			label = "Port Expander Reset";
-			gpios = <&gpio0 28 GPIO_ACTIVE_HIGH>;
-		};
-
-		batteryoutputenable: battery_output_enable {
-			label = "Battery Output Enable";
-			gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
-		};
-
-		do1mcu: do_1_mcu {
-			label = "Digital Output 1";
-			gpios = <&gpio0 12 GPIO_ACTIVE_HIGH>;
-		};
-
-		do2mcu: do_2_mcu {
-			label = "Digital Output 2";
-			gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
-		};
-
-		ain1sel: ain_1_current_measurement_selection {
-			label = "Analogue Input 1 Current/Voltage measurement select";
-			gpios = <&gpio_exp0 0 GPIO_ACTIVE_HIGH>;
-		};
-
-		ain2sel: ain_2_current_measurement_selection {
-			label = "Analogue Input 2 Current/Voltage measurement select";
-			gpios = <&gpio_exp0 1 GPIO_ACTIVE_HIGH>;
-		};
-
-		ain3sel: ain_3_current_measurement_selection {
-			label = "Analogue Input 3 Current/Voltage measurement select";
-			gpios = <&gpio_exp0 2 GPIO_ACTIVE_HIGH>;
-		};
-
-		ain4sel: ain_4_current_measurement_selection {
-			label = "Analogue Input 4 Current/Voltage measurement select";
-			gpios = <&gpio_exp0 3 GPIO_ACTIVE_HIGH>;
-		};
-
-		aina0: analog_input_selection_0 {
-			label = "Analogue Input Mux 0";
-			gpios = <&gpio_exp0 4 GPIO_ACTIVE_HIGH>;
-		};
-
-		aina1: analog_input_selection_1 {
-			label = "Analogue Input Mux 1";
-			gpios = <&gpio_exp0 5 GPIO_ACTIVE_HIGH>;
-		};
-
-		aux1: aux_input_1 {
-			label = "Aux Input 1";
-			gpios = <&gpio_exp0 6 GPIO_ACTIVE_HIGH>;
-		};
-
-		aux2: aux_input_2 {
-			label = "Aux Input 2";
-			gpios = <&gpio_exp0 7 GPIO_ACTIVE_HIGH>;
-		};
+	zephyr,user {
+		dinenable1-gpios = <&gpio1 5 GPIO_ACTIVE_HIGH>;
+		dinenable2-gpios = <&gpio1 10 GPIO_ACTIVE_HIGH>;
+		din1-gpios = <&gpio0 9 GPIO_PULL_UP>;
+		din2-gpios = <&gpio1 11 GPIO_PULL_UP>;
+		thermenable-gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+		analogenable-gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+		highsupplyenable-gpios = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+		expanderirq-gpios = <&gpio1 14 GPIO_PULL_UP>;
+		ain1-gpios = <&gpio0 2 GPIO_PULL_UP>;
+		ain2-gpios = <&gpio0 3 GPIO_PULL_UP>;
+		vref-gpios = <&gpio0 29 GPIO_PULL_UP>;
+		expanderreset-gpios = <&gpio0 28 GPIO_ACTIVE_HIGH>;
+		batteryoutputenable-gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+		do1mcu-gpios = <&gpio0 12 GPIO_ACTIVE_HIGH>;
+		do2mcu-gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+		ain1sel-gpios = <&gpio_exp0 0 GPIO_ACTIVE_HIGH>;
+		ain2sel-gpios = <&gpio_exp0 1 GPIO_ACTIVE_HIGH>;
+		ain3sel-gpios = <&gpio_exp0 2 GPIO_ACTIVE_HIGH>;
+		ain4sel-gpios = <&gpio_exp0 3 GPIO_ACTIVE_HIGH>;
+		aina0-gpios = <&gpio_exp0 4 GPIO_ACTIVE_HIGH>;
+		aina1-gpios = <&gpio_exp0 5 GPIO_ACTIVE_HIGH>;
+		aux1-gpios = <&gpio_exp0 6 GPIO_ACTIVE_HIGH>;
+		aux2-gpios = <&gpio_exp0 7 GPIO_ACTIVE_HIGH>;
 	};
```

Comment:
this should not be upstream, it is a bad practice to use DT like this.

Reply:
@mbolivar-ampere @galak

Reply:
I'm open to suggestions, I just want these to not be in `gpio-keys` :-)

Reply:
What is the best way to handle something like this then?

Reply:
Some of these entries seem to belong to actual devices, so they need their own compatible and a corresponding node. Here's also a good example https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/basic/custom_dts_binding.

I guess you can always keep this downstream and migrate your code, but I don't think upstream needs to deal with that.

Reply:
Oh I see now that `expanderirq-gpios` is already in `gpio_exp0`, `expanderreset-gpios` should just move there as well. The other I presume are all application specific so I guess there should be a binding in the application and the should move off upstream, is that what you are saying @gmarull?

Reply:
> Oh I see now that `expanderirq-gpios` is already in `gpio_exp0`, `expanderreset-gpios` should just move there as well. The other I presume are all application specific so I guess there should be a binding in the application and the should move off upstream, is that what you are saying @gmarull?

correct, also things like `highsupplyenable-gpios = <&gpio1 12 GPIO_ACTIVE_HIGH>;` seem to be a `regulator-fixed`. Other custom stuff should be dealt downstream in the application.

Reply:
Thank you both, lets try and get as much of this as possible supported upstream and then anything that doesn't fit will need to be downstream.

Reply:
I guess this will require some schematics

Reply:
- I converted the ones that sounded like pass transistors to regulator fixed (`din_enable_1 din_enable_2 therm_enable analog_enable high_supply_enable battery_output_enable`)
- port expander interrupt is already handled, port expander reset probably does not matter so I dropped it
- `v_ref` had a pullup, sounds like an input, probably should not have been there, dropped
- the remaining one are inputs or mux so they should be in the application (`din1-gpios din2-gpios ain1-gpios ain2-gpios do1mcu-gpios do2mcu-gpios ain1sel-gpios ain2sel-gpios ain3sel-gpios ain4sel-gpios aina0-gpios aina1-gpios aux1-gpios aux2-gpios`)

The other board I left the id pins under `zephyr,user`, does not bother me but lmk what to do.